### ISSUE TITLE
Support java.time Instant and java.time Clock

### ISSUE THREAD

**Issue by Marthym** [jwtk/jjwt#286]
It should be a great thing to be able to support java.time.Instant and java.time.Clock instant of Date and io.jsonwebtoken.Clock.

``` java
Jwts.builder()
       .setSubject(user.username())
       .setId(tokenId.toString())
       .setExpiration(Instant.now().plus(expiration))
       .setIssuedAt(Date.from(now))
       .addClaims(claimsBldr.build())
       .signWith(SignatureAlgorithm.HS512, signingKey)
       .compact();
```

```
Jwts.parser()
       .setSigningKey(signingKey)
       .setClock(Clock.systemUTC())
       .parseClaimsJws(token.replace(PREFIX, ""))
       .getBody();
```

**Comment by lhazlewood** [comment]
This depends on #308

**Comment by lhazlewood** [comment]
Closing as duplicate of #235.  Please #235 for work/discussion.
